NetworkingTests/Tests/SessionManagerSpec.swift

```diff
@@ -0,0 +1,473 @@
+                }
+                
+                it("returns the current user") {
+                    sessionManager.login(user: UserMock())
+                    expect(sessionManager.sessionToken).to(equal(UserMock().sessionToken))
+                }
+
+            }
+            
+        }
+        
+        describe("#update") {
+            
+            describe("when there is no current session") {
+                
+                beforeEach {
+                    keychainService = KeychainServiceMock()
+                    sessionManager = SessionManager(keychainService: keychainService)
+                    sessionManager.bootstrap()
+                }
+                
+                it("throws an assertion") {
+                    expect(sessionManager.update(user: UserMock())).to(throwAssertion())
+                }
+                
+            }
+            
+            describe("when there is a current session") {
+                
+                let updatedSessionToken = "updated-fake-session-token"
+                
+                beforeEach {
+                    keychainService = KeychainServiceMock()
+                    keychainService.set(value: UserMock().sessionToken!,
+                                        forKey: SessionManagerSpec.CurrentSessionTokenPersistanceKey)
+                    sessionManager = SessionManager(keychainService: keychainService)
+                    sessionManager.bootstrap()
+                }
+                
+                it("sends no session signal") {
+                    sessionManager.sessionSignal.observeValues { _ in
+                        fail()
+                    }
+                    
+                    sessionManager.update(user: UserMock())
+                }
+                
+                it("sends the updated authenticated user in user signal") { waitUntil { done in
+                    sessionManager.userSignal.observeValues {
+                        switch $0 {
+                        case .none: fail()
+                        case .some(let user):
+                            expect(user.sessionToken).to(equal(updatedSessionToken))
+                            done()
+                        }
+                    }
+                    
+                    var updatedUser = UserMock()
+                    updatedUser.sessionToken = updatedSessionToken
+                    sessionManager.update(user: updatedUser)
+                }}
+                
+                it("updates the current user") {
+                    var updatedUser = UserMock()
+                    updatedUser.sessionToken = updatedSessionToken
+                    sessionManager.update(user: updatedUser)
+                    expect(sessionManager.currentUser!.sessionToken).to(equal(updatedSessionToken))
+                }
+                
+                it("does not update the session token") {
```

Comment:
And should this happen? I don't think is okay someone can change the token of a user and not change the session token. And the token should be valid, from a session. I don't think the user is meant to have the token itself. And so this wouldn't happen. You could just change for example the name of the user and not have any inconsistencies.